save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt

```diff
@@ -272,30 +271,47 @@ class LnkUserOrganizationController(
     )
     @ApiResponse(responseCode = "200", description = "Successfully fetched list of users")
     @ApiResponse(responseCode = "404", description = "Requested organization doesn't exist")
+    @Suppress("TOO_LONG_FUNCTION")
     fun getAllUsersNotFromOrganizationWithNamesStartingWith(
         @PathVariable organizationName: String,
         @RequestParam prefix: String,
         authentication: Authentication,
-    ): List<UserInfo> {
-        if (prefix.isEmpty()) {
-            return emptyList()
-        }
-        val organization = organizationService.findByName(organizationName)
-            ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
-        val organizationUserIds = lnkUserOrganizationService.getAllUsersAndRolesByOrganization(organization)
-            .map { (user, _) ->
-                user.requiredId()
-            }.toSet()
-        // first we need to get users with exact match by name
-        val exactMatchUsers = lnkUserOrganizationService.getNonOrganizationUsersByName(prefix, organizationUserIds)
-        // and then we need to get those whose names start with `prefix`
-        val prefixUsers = lnkUserOrganizationService.getNonOrganizationUsersByNamePrefix(
-            prefix,
-            organizationUserIds + exactMatchUsers.map { it.requiredId() },
-            PAGE_SIZE - exactMatchUsers.size,
-        )
-        return (exactMatchUsers + prefixUsers).map { it.toUserInfo() }
-    }
+    ): Mono<List<UserInfo>> = Mono.just(organizationName)
+        .filter {
+            prefix.isNotEmpty()
+        }
+        .flatMap {
+            organizationService.findByName(it).toMono()
+        }
+        .switchIfEmptyToNotFound {
+            "No organization with name $organizationName was found."
+        }
+        .map { organization ->
+            lnkUserOrganizationService.getAllUsersAndRolesByOrganization(organization)
+        }
```

Comment:
Can be converted to Flux using `flatMapMany` and `Flux.fromIterable`

Reply:
The point is that later we will need all the list we get here because we pass it later to `LnkUserOrganizationService`

Reply:
Ok, haven't noticed